repose-aggregator/components/filters/url-extractor-to-header/src/main/scala/org/openrepose/filters/urlextractortoheader/UrlExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,93 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.urlextractortoheader
+
+import java.net.URL
+import javax.inject.{Named, Inject}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.urlextractortoheader.config.UrlExtractorToHeaderConfig
+
+import scala.collection.JavaConverters._
+import scala.util.matching.Regex
+
+@Named
+class UrlExtractorToHeaderFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[UrlExtractorToHeaderConfig] with LazyLogging {
+
+  import UrlExtractorToHeaderFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var extractions: Iterable[Extraction] = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("URL Extractor to Header filter initializing...")
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info("Initializing URL Extractor to Header filter using config {}", configurationFile)
+    val xsdUrl: URL = getClass.getResource(SCHEMA_FILE_NAME)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFile, xsdUrl, this, classOf[UrlExtractorToHeaderConfig])
+
+    logger.trace("URL Extractor to Header filter initialized.")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+
+    extractions.foreach { extraction =>
+      (extraction.urlRegex.findFirstIn(mutableHttpRequest.getRequestURI), extraction.defaultValue) match {
+        case (Some(extraction.urlRegex(headerValue)), _) => mutableHttpRequest.addHeader(extraction.headerName, headerValue)
+        case (None, Some(defaultValue)) => mutableHttpRequest.addHeader(extraction.headerName, defaultValue)
+        case (None, None) => // don't add a header
```

Comment:
I'm torn between leaving our third case explicit, and future proofing by using case (_, _).